python/restate/server_context.py

```diff
@@ -74,6 +74,7 @@ async def await_point():
                         self.state = "rejected"
                         raise t
                 case "fulfilled":
+                    assert self.value is not None
```

Comment:
Hi, I implemented quite aggressive type asserts, which will help maintainability down the road.
All tests pasts, however I'm wondering, is there a case where the value could be `None` if state is `fulfilled`?
If so, this line should be a type cast instead of an assert.